### DISCUSSION TITLE
:art: Billing billing/billing-entries - extend endpoint

### DISCUSSION THREAD

**Discussion by BartoszWoszczak** [allegro/allegro-api#7666] [Suggestions - resolved]
Dzień dobry,

Czy moglibyście rozszerzyć zawartość endpointu https://api.{environment}/billing/billing-entries?

Chodzi tutaj o zawieranie dokładnej informacji względem typu operacji. Fajnie by to było wywoływane tylko, jeżeli uyżytkownik będzie tego potrzebował tj.

Dodajemy parametr do GET `showDetails` o wartości `true/1/yes` który pozwoli na dogranie kontekstu każdej operacji. Idealnym przykładem, jest problem kiedy masz jedno zamówienie, kilka paczek i listów przewozowych ale nie wiesz która paczka to który koszt dokładnie, przez co odczyt z tej listy nie może odbyć się w formie jakiej byśmy wymagali. 

Dane końcowe mogą wyglądać tak:

```json
{
  "billingEntries": [
    {
      "id": "09f0b4cc-7880-11e9-8f9e-2a86e4085a59",
      "occurredAt": "2019-05-08T09:45:32.818Z",
      "type": {
        "id": "LIS",
        "name": "Listing fee"
      },
      "offer": {
        "id": "12345678",
        "name": "offer name"
      },
      "value": {
        "amount": "100.00",
        "currency": "PLN"
      },
      "tax": {
        "percentage": "0",
        "annotation": "OUT_OF_SCOPE"
      },
      "balance": {
        "amount": "100.00",
        "currency": "PLN"
      },
      "order": {
        "id": "09f0b4cc-7880-11e9-8f9e-2a86e4085a59"
      },
      "shipping": {
        "id": "... id to shipping",
        "name": "Name of delivery like InPost, Paczkomat, DPD, UPS and etc.",
        "letterNumber": "Package letter number to find in external system elements connected to this package"
      }
    }
  ]
}
```

Oczywiście to sugestia rozwiązania, ale wydaje się do zrealizowania w ciągu kilku godzin bez jakichś wielkich zmian wkodzie. Bez parametru wszystko odbywa się po staremu, parametr odpala procedure uzupełnienia danych z odpowiednimi typami.

Pozdrawiam i zachęcam do współpracy w razie braku developerów do takich prostych tematów :)

**Comment by MartaNowaczyk** [comment]
Po konsultacjach z zespołem odpowiedzialnym za ten endpoint mam informację, że nie planujemy wdrożyć takiej zmiany. W takim wypadku sugestię oznaczam jako "Suggestions - rejected".

**Reply by arekpietrzyk** [reply]
@MartaNowaczyk a czy można liczyć na chociaż rozszerzenie zwracanych w tym endpoincie informacji do poziomu informacji zawartych w pliku raportu CSV?

Mam tu na myśli przede wszystkim dodatkowe informacje związane z dopłatami, jak na przykładzie poniżej:

"Numer nadania: xxxyyyzzz, Kod: DPD_P1_KURIER, Identyfikator zamówienia: x-y-z-y, Szczegóły: Allegro Kurier DPD; Order value = 110.00 PLN - w ramach Uslugi Allegro Smart"

Dla tej pozycji z pliku CSV, w API widać jedynie kod typu: HB1

**Reply by MartaNowaczyk** [reply]
@arekpietrzyk przekazałam Twoje rozwiązanie do odpowiedniego zespołu. Na czas ponownej weryfikacji zmieniam status "Suggestion - verification".

**Reply by arekpietrzyk** [reply]
Dziękuję. Najważniejszy i brakujący element to "DPD_P1_KURIER".

Dla przykładu z innego zamówienia, kod bilingowy: DTR, brakujący kod uzupelniający "DPD_N2_PRZES_NSTAND".

**Reply by kontakt-electrobit** [reply]
Podłączam się do prośby o dodanie dodatkowych informacji do opłat. 
Obecnie poprzez API nie ma informacji o szczególe opłat dodatkowych dla przesyłek. W przypadku np. DPD opłata 4,99 to zarówno DPD_P2_KURIER_POBRANIE jak i DPD_N1_PODJAZD.

**Reply by arekpietrzyk** [reply]
@MartaNowaczyk chcialbym poprosic o sprawdzenie w zespole w kwestii dodania tych brakujacych detali wzgledem pliku CSV, gdyz trzeba podjac decyzje o czekaniu na API lub inwestowaniu w parser CSV - ktory pojdzie do kosza jak rozszerzycie API.

To raptem kilka pol ktore developerzy po Waszej stronie maja dostepne.

**Comment by kontakt-electrobit** [comment]
Dzień dobry
Czy jest jakaś aktualizacja w sprawie wdrożenia zmiany?
Informacje o szczegółach dopłaty do przesyłek są konieczne zwłaszcza przy zamówieniach z zagranicznych marketplace, gdzie ze względu na zmienne kursy walut nie można stworzyć własnego mapowania koszt=rodzaj dopłaty.

**Reply by MartaNowaczyk** [reply]
Na ten moment nie mamy tego w planach, jeśli sytuacja się zmieni damy znać.

**Comment by MartaNowaczyk** [comment]
[Wdrożyliśmy dziś zmianę](https://github.com/allegro/allegro-api/issues/11330), która zwraca szczegóły dla typów opłat, tak jak w "Rozliczeniach Allegro".
